snuba_sdk/entity.py

```diff
@@ -16,27 +16,39 @@ class InvalidEntity(Exception):
 class Entity(Expression):
     name: str
     alias: Optional[str] = None
-    sample: Optional[Union[int, float]] = None
+    sample: Optional[float] = None
 
     def validate(self) -> None:
         # TODO: There should be a whitelist of entity names at some point
         if not isinstance(self.name, str) or not entity_name_re.match(self.name):
             raise InvalidEntity(f"'{self.name}' is not a valid entity name")
 
         if self.sample is not None:
-            if not isinstance(self.sample, (int, float)):
-                raise InvalidEntity(
-                    "sample must be a float between 0 and 1 or an integer greater than 1"
-                )
-            elif isinstance(self.sample, float):
-                if self.sample < 0.0 or self.sample > 1.0:
-                    raise InvalidEntity(
-                        "float samples must be between 0.0 and 1.0 (%age of rows)"
-                    )
-            elif isinstance(self.sample, int):
-                if self.sample < 1:
-                    raise InvalidEntity("int samples must be at least 1 (# of rows)")
+            if not isinstance(self.sample, float):
+                raise InvalidEntity("sample must be a float")
+            elif self.sample <= 0.0:
+                raise InvalidEntity("samples must be greater than 0.0")
 
         if self.alias is not None:
             if not isinstance(self.alias, str) or not self.alias:
                 raise InvalidEntity(f"'{self.alias}' is not a valid alias")
+
+
+# TODO: This should be handled by the users of the SDK, not the SDK itself.
+ENTITY_TIME_COLUMNS = {
```

Comment:
Do you have a plan on how to move these out of the sdk? I think having these hardcoded is a pretty serious issue as it is not going to be required for the migration only. Also we will forget to keep them up to date. 

From the old query language perspective these columns are dataset specific, not entity specific (they do not know entities). What about providing a mapping dataset to timestamp column to the sdk when querying ? We could make it transparent into utils/snuba.py

Reply:
So there are two reasons why this won't get stagnant: firstly, this is only necessary for legacy -> snql conversions. Once that need goes away, so does this. Secondly, I plan to rewrite all of this when I add the data model component to the SDK, so this will disappear at that point as well.

Having Sentry provide the mapping is fine, but that data will have to live in 3 places then: the SDK and Snuba (for tests) as well as Sentry. Also is it worth the added complexity for logic that is slated to be written before the end of the quarter?

Reply:
> firstly, this is only necessary for legacy -> snql conversions.

To confirm. We do not plan to add mandatory conditions validation in the sdk, right? 

> Secondly, I plan to rewrite all of this when I add the data model component to the SDK, so this will disappear at that point as well.

This sounds a good plan to me.

Reply:
> To confirm. We do not plan to add mandatory conditions validation in the sdk, right? 

I do plan to add the concept of mandatory condition validation to the SDK, but it would be up to user to define what the actual conditions should be. Similar to how the Snuba Entity has a required_columns field and knows how to validate that field, but the specifics of which field should be validated is left to the user of that class.